src/eip4824.ts

```diff
@@ -35,28 +35,54 @@ router.get('/:space', async (req, res) => {
 
 router.get('/:space/members', async (req, res) => {
   let space: any = {};
+  let uniqueVotersResult: any = {};
 
   try {
-    space = await getSpace(req.params.space);
+    const spaceId = req.params.space;
+    const cursor = req.query.cursor || null;
+    // Default pageSize is set to 100, this is the query limit.
 
+    space = await getSpace(spaceId);
     if (!space.verified) return res.status(400).json({ error: 'INVALID' });
+
+    uniqueVotersResult = await getUniqueVotersForSpace(spaceId, cursor);
   } catch (e) {
     return res.status(404).json({ error: 'NOT_FOUND' });
   }
 
-  const members = [...space.admins, ...space.moderators, ...space.members].map(
-    member => ({
-      type: 'EthereumAddress',
-      id: member
-    })
-  );
-
-  return res.json({
+  // Combine all members and map them to JSON strings for uniqueness check
+  const allMembersJSON = [
+    ...space.admins.map(admin =>
+      JSON.stringify({ type: 'EthereumAddress', id: admin })
+    ),
+    ...space.moderators.map(moderator =>
+      JSON.stringify({ type: 'EthereumAddress', id: moderator })
+    ),
+    ...space.members.map(member =>
+      JSON.stringify({ type: 'EthereumAddress', id: member })
+    ),
+    ...uniqueVotersResult.voters.map(voter =>
+      JSON.stringify({ type: 'EthereumAddress', id: voter })
+    )
+  ];
+
+  // Convert the array of JSON strings to a Set to remove duplicates, then back to an array
+  const uniqueMembersJSON = [...new Set(allMembersJSON)];
```

Comment:
This is not ideal because you may not always return 100 results, I would rather make a query on the db to enforce that we don't get any duplicated addresses from admin, moderator or authors (members).

Reply:
I address this by querying the DB with this query

`query = `
    SELECT address
    FROM (
      (SELECT DISTINCT JSON_UNQUOTE(JSON_EXTRACT(settings, '$.admins[*]')) AS address 
       FROM spaces 
       WHERE id = ?)
      UNION
      (SELECT DISTINCT JSON_UNQUOTE(JSON_EXTRACT(settings, '$.members[*]')) AS address 
       FROM spaces 
       WHERE id = ?)
      UNION
      (SELECT DISTINCT JSON_UNQUOTE(JSON_EXTRACT(settings, '$.moderators[*]')) AS address 
       FROM spaces 
       WHERE id = ?)
      UNION
      (SELECT DISTINCT voter AS address
       FROM votes
       WHERE space = ? AND (? IS NULL OR voter > ?)
    ) AS combined_addresses
    ORDER BY address
    LIMIT ?
`;
`
And the last address after the limit reached is considered as cursor